substrate-node/pallets/pallet-tft-bridge/src/migrations/v2.rs

```diff
@@ -0,0 +1,120 @@
+use crate::*;
+use frame_support::log::{debug, info};
+use frame_support::{traits::Get, traits::OnRuntimeUpgrade, weights::Weight};
+use parity_scale_codec::Encode;
+use sp_std::marker::PhantomData;
+
+#[cfg(feature = "try-runtime")]
+use sp_runtime::codec::Decode;
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+pub struct MigrateBurnTransactionsV2<T: Config>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for MigrateBurnTransactionsV2<T> {
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        info!("current pallet version: {:?}", PalletVersion::<T>::get());
+        if PalletVersion::<T>::get() != types::StorageVersion::V1 {
+            return Ok(Vec::<u8>::new());
+        };
+
+        let burn_transactions_count: u64 =
+            migrations::types::v1::BurnTransactions::<T>::iter().count() as u64;
+        info!(
+            "🔎 MigrateBurnTransactionsV2 pre migration: Number of existing burn transactions {:?}",
+            burn_transactions_count
+        );
+
+        let executed_burn_transactions_count: u64 =
+            migrations::types::v1::ExecutedBurnTransactions::<T>::iter().count() as u64;
+        info!(
+            "🔎 MigrateBurnTransactionsV2 pre migration: Number of existing executed burn transactions {:?}",
+            executed_burn_transactions_count
+        );
+
+        info!("👥  TFT-BRIDGE pallet to V1 passes PRE migrate checks ✅",);
+        return Ok(Vec::<u8>::new());
+    }
+
+    fn on_runtime_upgrade() -> Weight {
+        if PalletVersion::<T>::get() == types::StorageVersion::V1 {
+            migrate_burn_transactions::<T>()
+        } else {
+            info!(" >>> Unused TFT-BRIDGE pallet V2 migration");
+            Weight::zero()
+        }
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(_pre_burn_transactions_count: Vec<u8>) -> Result<(), &'static str> {
+        info!("current pallet version: {:?}", PalletVersion::<T>::get());
+        if PalletVersion::<T>::get() != types::StorageVersion::V2 {
+            return Ok(());
+        }
+        let burn_transactions_count: u64 = BurnTransactions::<T>::iter().count() as u64;
+        info!(
+            "🔎 MigrateBurnTransactionsV2 post migration: Number of existing burn transactions {:?}",
+            burn_transactions_count
+        );
+
+        let executed_burn_transactions_count: u64 =
+            ExecutedBurnTransactions::<T>::iter().count() as u64;
+        info!(
+            "🔎 MigrateBurnTransactionsV2 post migration: Number of existing executed burn transactions {:?}",
+            executed_burn_transactions_count
+        );
+
+        Ok(())
+    }
+}
+
+pub fn migrate_burn_transactions<T: Config>() -> frame_support::weights::Weight {
+    info!(" >>> Migrating burn transactions storage...");
+
+    let mut read_writes = 0;
+
+    BurnTransactions::<T>::translate::<super::types::v1::BurnTransaction<T::BlockNumber>, _>(
```

Comment:
The migration code only runs when the storage version matches the expected one, and it ensures that the migration does not repeat itself. I used try-runtime to test the migration and it verified that the migration was idempotent.

Reply:
I mean if we run multiple migrations V1->V2 and V2->V3 in same block and modify `BurnTransaction` type each time we will have a type incompatibility issue because same scope.

Ex:
~~~rust
type Migrations = (
    ...
    pallet_tft_bridge::migrations::v2::MigrateBurnTransactionsV2<Runtime>,
    pallet_tft_bridge::migrations::v3::MigrateBurnTransactionsAgainV3<Runtime>,
    ...
);
~~~

This is just safer and good practice to avoid such conflict.
Not sure explanation is clear?

Reply:
Yes, I see.

Reply:
Done